### PR TITLE
Ingenico Commands Pull Request

### PR DESCRIPTION
**Pull request by stevenlestatan**
Hi Russell,

In here forked repository are the changes in your current java-sdk for the Ingenico commands purposes.

Kind regards,
Steven Tan of Serino Team

### CODE REVIEW

File: src/main/java/com/global/api/entities/exceptions/GatewayTimeoutException.java [hunk 1]
```diff
@@ -53,9 +53,9 @@ public void setTransmissionTime(String transmissionTime) {
     }
 
     public GatewayTimeoutException() {
-        super("The gateway did not respond within the given timeout.");
+        super("The gateway did not response with given timeout.");
```
Comment (russelleverett) [inline src/main/java/com/global/api/entities/exceptions/GatewayTimeoutException.java:56]: This was correct the way it was

File: src/main/java/com/global/api/gateways/Gp3DSProvider.java [hunk 2]
```diff
@@ -139,7 +139,7 @@ else if(paymentMethod instanceof RecurringPaymentMethod) {
                 hashValue = storedCard.getCustomerKey();
 
                 cardDetailElement.set("payer_reference", storedCard.getCustomerKey())
-                        .set("payment_method_reference", storedCard.getKey());
+                        .set("payment_reference", storedCard.getKey());
```
Comment (russelleverett) [inline src/main/java/com/global/api/gateways/Gp3DSProvider.java:142]: This was ALSO correct the way it was

File: pom.xml [hunk 3]
```diff
@@ -1,141 +1,146 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project xmlns="http://maven.apache.org/POM/4.0.0"
-         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
```
[the remark this answers is not in the document]
Comment (stevenlestatan) [reply pom.xml:3]: As I checked and compared the pom.xml from my forked repo to your pom.xml from your master branch. The whole file just adjusted their indentions, nothing had been edit but added dependency only that is the artifactId of jSerialComm. Should I adjust indentions of each line?

File: src/main/java/com/global/api/builders/AuthorizationBuilder.java [hunk 4]
```diff
@@ -44,6 +44,8 @@
     private String customerIpAddress;
     private String cvn;
     private DccRateData dccRateData;
+//    private DccProcessor dccProcessor;
```
[the remark this answers is not in the document]
Comment (stevenlestatan) [reply src/main/java/com/global/api/builders/AuthorizationBuilder.java:47]: Actually, I have nothing to do with this class. I didn't enhanced this from the time I cloned the forked repo from your master branch.

File: src/main/java/com/global/api/builders/ManagementBuilder.java [hunk 5]
```diff
@@ -144,6 +145,9 @@ public BigDecimal getTaxAmount() {
     public TaxType getTaxType() {
         return taxType;
     }
+    public boolean isTerminalError() {
```
[the remark this answers is not in the document]
Comment (stevenlestatan) [reply src/main/java/com/global/api/builders/ManagementBuilder.java:148]: Same with here, I don't really know how did that come there. I really have nothing to do with this.

File: src/main/java/com/global/api/builders/TransactionBuilder.java [hunk 6]
```diff
@@ -28,7 +28,6 @@
     protected int systemTraceAuditNumber;
     protected String uniqueDeviceId;
     protected TransactionMatchingData transactionMatchingData;
-    protected boolean terminalError;
```
[the remark this answers is not in the document]
Comment (stevenlestatan) [reply src/main/java/com/global/api/builders/TransactionBuilder.java:31]: Sorry, I believe in here also. I don't really know why they're moved. Nothing to do with this class also.

File: src/main/java/com/global/api/entities/enums/PaymentMethodType.java [hunk 7]
```diff
@@ -11,7 +11,8 @@
     Cash,
     ACH,
     Gift,
-    Recurring;
+    Recurring,
+    Other;
```
[the remark this answers is not in the document]
Comment (stevenlestatan) [reply src/main/java/com/global/api/entities/enums/PaymentMethodType.java:15]: I did not add that. From the time I cloned the forked repo, it's already there. As I checked my sourcetree commits and pushed, and from remembering all what I did for the forked repo of java-sdk, there's nothing enhancement I made here.

File: src/main/java/com/global/api/entities/exceptions/GatewayException.java [hunk 8]
```diff
@@ -1,15 +1,26 @@
 package com.global.api.entities.exceptions;
 
+import com.global.api.gateways.events.IGatewayEvent;
```
[the remark this answers is not in the document]
Comment (stevenlestatan) [reply src/main/java/com/global/api/entities/exceptions/GatewayException.java:3]: Sorry, I didn't update this. I didn't enhance any of the exceptions class.

File: src/main/java/com/global/api/entities/exceptions/GatewayTimeoutException.java (hunk 1 above)
Comment (stevenlestatan) [reply src/main/java/com/global/api/entities/exceptions/GatewayTimeoutException.java:56]: Nothing to do with this. I my not the one who optimize the given parameter string of that base class.

File: src/main/java/com/global/api/gateways/Gp3DSProvider.java (hunk 2 above)
Comment (stevenlestatan) [reply src/main/java/com/global/api/gateways/Gp3DSProvider.java:142]: Same goes here.

**Comment by stevenlestatan** [comment]
Hi Russell,

 

As I checked again the whole master branch of java-sdk from the github, nothing has been changed from March 17 that why as I compared again the whole repo from my forked repo, there’s nothing new to pull. I have already replied with all the comments you in for the PR I made.

 

Kindly see it.

 

Thank you so much!

 

Kind regards,

Steven

**Pull request closed without merging**